### PR TITLE
[Java] interfaces new exercise

### PR DESCRIPTION
**Pull request by mikedamay**
All files complete except java/config.json which I will update before mergoing

### CODE REVIEW

File: languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class RemoteControlCarTest {
+    @Test
+    public void race() {
+        ProductionRemoteControlCar productionCar = new ProductionRemoteControlCar();
+        ExperimentalRemoteControlCar experimentalCar = new ExperimentalRemoteControlCar();
+        TestTrack.race((RemoteControlCar)productionCar);
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java:14]: to respect track style guidelines, please a whitespace between cast and variable name
example `(RemoteControCar) productionCar`

File: languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        prc1.setNumberOfVictories(3);
+        prc2.setNumberOfVictories(2);
+        List<ProductionRemoteControlCar> rankings = TestTrack.getRankedCars(prc1, prc2);
+        assertSame(prc1, rankings.get(1));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void ensureCarsAreComparables() {
+        assertTrue(RemoteControlCar.class.isAssignableFrom(ProductionRemoteControlCar.class) );
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java:35]: to respect track style guidelines, please remove the whitespace before the last `)`

**Review by mirkoperillo — CHANGES_REQUESTED** [review]
@mikedamay thank you, only some minor fixes and then the PR will be ok.

File: languages/java/exercises/concept/interfaces/.docs/hints.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+- Setters and getters behave in the same way as any other method.
+
+## 3. Allow the production cars to be ranked
+
+- See [this discussion][sort] of sorting.
+- See [here][comparable] for default comparison of objects.
+
+[interfaces]: https://docs.oracle.com/javase/tutorial/java/concepts/interface.html
+[interface-property]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/interface-properties
```
[the remark this answers is not in the document]
Comment (mikedamay) [reply languages/java/exercises/concept/interfaces/.docs/hints.md]: Done

File: languages/java/exercises/concept/interfaces/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## 2. Enable the distance travelled by different models on the test track to be compared
+
+Please add a method to the `RemoteControlCar` interface to encapsulate the behavior of the `getDistanceTravelled()` method for the two types of car.
+
+```java
+ProductionRemoteControlCar prod = new ProductionRemoteControlCar();
+TestTrack.race(prod);
+ExperimentalRemoteControlCar exp = new ExperimentalRemoteControlCar();
+TestTrack.race(exp);
+prod.getDistanceTravelled()
```
[the remark this answers is not in the document]
Comment (mikedamay) [reply languages/java/exercises/concept/interfaces/.docs/instructions.md]: Done

File: languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java (hunk 1 above)
Comment (mikedamay) [reply languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java]: Done

File: languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java (hunk 2 above)
Comment (mikedamay) [reply languages/java/exercises/concept/interfaces/src/test/java/RemoteControlCarTest.java]: Done

**Comment by mikedamay** [comment]
All review pionts addressed will update java/config.json before merging

**Review by mirkoperillo — APPROVED** [review]

**Comment by mirkoperillo** [comment]
@mikedamay thank you, time to merge the PR

**Pull request merged**